components/brave_rewards/resources/extension/brave_rewards/background/publisherVisit.ts

```diff
@@ -44,6 +81,23 @@ const handleGreaselionError = (tabId: number, mediaType: string, data: Greaselio
   console.error(`Greaselion error: ${data.errorMessage}`)
 }
 
+const handleMakeAPIRequest = (data: MakeAPIRequest, sendResponse: any) => {
+  if (!data || !data.url || !data.init) {
+    return
+  }
+
+  fetch(data.url, data.init)
+    .then(response => {
+      if (!response.ok) {
+        throw new Error(`API request failed: ${response.statusText} (${response.status})`)
+      }
+
+      return response.json()
+    })
+    .then(responseData => sendResponse(responseData))
+    .catch(error => sendResponse(error))
```

Comment:
It looks like we're sending both errors and valid responses through the "response" field into the content script. Does the content script know how to tell the difference? Perhaps the message to the content script should include both "response" and "error" fields?

Reply:
Great point, fixed.